src/html5/parser.rs

```diff
@@ -3905,4 +3894,56 @@ mod test {
         binding.set_node_named_id(NodeId(4), "otherid");
         assert!(binding.get_node_by_named_id("myid").is_none());
     }
+
+    #[test]
+    fn tests2_dat_720() {
+        let document = parse("<!DOCTYPE html>X</html>X");
+
+        assert_same(
+            &document,
+            r#"
+            | <!DOCTYPE html>
+            | <html>
+            |   <head>
+            |   <body>
+            |     "XX"
+            "#,
+        );
+    }
+
+    #[test]
+    fn tests2_dat_731() {
+        let document = parse("<!DOCTYPE html>X</html> ");
+
+        assert_same(
+            &document,
+            r#"
+            | <!DOCTYPE html>
+            | <html>
+            |   <head>
+            |   <body>
+            |     "X "
+            "#,
+        );
```

Comment:
The fact that this test passes suggests that we're not handling it right in the other test code, making it a false negative.